### ISSUE TITLE
Issue when rejoining an inactive validator

### ISSUE THREAD

**Issue by TomVasile** [casper-network/casper-node#1558]
### Overview:

TL;DR: If we restart a node after it is marked inactive but prior to ejection, then rebid it after ejection, it fails to rejoin.

Fails consistently in case 1 ( labeled in my branch ) and passes consistently in case 2 ( labeled in my branch ).
- Case 1 implements what was proposed in the issue. ie. restarting the node prior to verifying ejection
- Case 2 is basically the same as the current test, restarts the node after verified ejection

### Steps to repro:
```
cd "$NCTL/sh/scenarios"
nctl-assets-teardown
nctl-assets-setup chainspec_path="$NCTL/sh/scenarios/chainspecs/itst13.chainspec.toml.in"
nctl-start
sleep 90
bash -i itst13.sh
```

### Links:
Test description: https://app.zenhub.com/workspaces/engineering-60953fafb1945f0011a3592d/issues/casper-network/governance/14

My branch: https://github.com/TomVasile/casper-node/tree/governance-issue-14

Test in question: [itst13.sh](https://github.com/TomVasile/casper-node/blob/governance-issue-14/utils/nctl/sh/scenarios/itst13.sh)

**Comment by afck** [comment]
If I set

```toml
standstill_timeout = '30s'
```

in the `config.toml` it passes for me. In general it doesn't make sense to have a timeout that's longer than an era (41 seconds in this test). Otherwise the joining node could start era 4 just when the others move on to 5 and make no progress for a long time.

**Comment by afck** [comment]
@fizyk20 also saw the test fail with an unexpected proposed block:
I realized I described the feature the wrong way. You're really only excluded from the leader sequence if you were inactive in the _previous_ era. Not in general if you're about to be evicted. So: If the node was inactive in era 4 then it doesn't get to propose blocks in era 5. However, if it sends any message in era 5, it _does_ get to propose blocks in era 6, even if it'll be evicted in era 7.

So I think the test should be modified: The era length should be increased, the standstill timeout decreased, and `auction_delay` set to 1 (like in mainnet).

**Comment by afck** [comment]
@fizyk20 and I probably finally found the root cause. This branch should work: https://github.com/afck/casper-node/tree/governance-issue-14 (Testing right now…)

It caused the inactive and faulty validators to be reversed when initializing the next era, but _only_ in the existing nodes. So the re-joining node found itself being accused of an equivocation it hadn't actually committed and asked for evidence. Of course the other nodes didn't have any evidence, so the joining node was stalled.

We'll improve the code to make this mix-up less likely to happen again (stronger types, more tests) next week, but if the tests pass I will make a PR with the fix today.
